modules/PanelApp.py

```diff
@@ -0,0 +1,227 @@
+        )
+    ]
+
+    def __init__(self):
+        self.spark = SparkSession.builder.appName('panelapp_parser').getOrCreate()
+
+    def generate_panelapp_evidence(
+        self,
+        input_file: str,
+        output_file: str,
+    ) -> None:
+        # Filter and extract the necessary columns.
+        panelapp_df = (
+            self.spark.read.csv(input_file, sep=r'\t', header=True)
+            .filter(
+                ((col('List') == 'green') | (col('List') == 'amber')) &
+                (col('Panel Version') > 1) &
+                (col('Panel Status') == 'PUBLIC')
+            )
+            .select('Symbol', 'Panel Id', 'Panel Name', 'Mode of inheritance', 'Phenotypes')
+        )
+
+        # Fix typos and formatting errors which interfere with phenotype splitting.
+        for regexp, replacement in self.PHENOTYPE_BEFORE_SPLIT_RE.items():
+            panelapp_df = panelapp_df.withColumn('Phenotypes', regexp_replace(col('Phenotypes'), regexp, replacement))
+
+        # Split and explode the phenotypes.
+        panelapp_df = (
+            panelapp_df
+            .withColumn('cohortPhenotypes', array_distinct(split(col('Phenotypes'), ';')))
+            .withColumn('phenotype', explode(col('cohortPhenotypes')))
+            .drop('Phenotypes')
+        )
+
+        # Remove specific patterns and phrases which will interfere with ontology extraction and mapping.
+        for regexp in self.PHENOTYPE_AFTER_SPLIT_RE:
+            panelapp_df = panelapp_df.withColumn('phenotype', regexp_replace(col('phenotype'), f'({regexp})', ''))
+
+        # Extract ontology information, clean up and filter the split phenotypes.
+        panelapp_df = (
+            panelapp_df
+
+            # Extract Orphanet/MONDO/HP ontology identifiers and remove them from the phenotype string.
+            .withColumn('ontology_namespace', regexp_extract(col('phenotype'), self.OTHER_RE, 1))
+            .withColumn('ontology_namespace', regexp_replace(col('ontology_namespace'), 'OrphaNet: ORPHA', 'ORPHA'))
+            .withColumn('ontology_id', regexp_extract(col('phenotype'), self.OTHER_RE, 2))
+            .withColumn(
+                'ontology',
+                when(
+                    (col('ontology_namespace') != '') & (col('ontology_id') != ''),
+                    concat(col('ontology_namespace'), lit(':'), col('ontology_id'))
+                )
+            )
+            .withColumn('phenotype', regexp_replace(col('phenotype'), f'({self.OTHER_RE})', ''))
+
+            # Extract OMIM identifiers and remove them from the phenotype string.
+            .withColumn('omim_id', regexp_extract(col('phenotype'), self.OMIM_RE, 2))
+            .withColumn('omim', when(col('omim_id') != '', concat(lit('OMIM:'), col('omim_id'))))
+            .withColumn('phenotype', regexp_replace(col('phenotype'), f'({self.OMIM_RE})', ''))
+
+            # Choose one of the ontology identifiers, keeping OMIM as a priority.
```

Comment:
As discussed elsewhere, this doesn't really have an effect in this case because OMIM and other ontologies don't co-occur in the same record. OMIM is the de factor standard for the PanelApp, so it seems to make sense to prefer it